### ISSUE TITLE
Is there any way to stop segment anything 2 from reinstalling itself every time its called?

### ISSUE THREAD

**Issue by casabelle42** [autodistill/autodistill-grounded-sam-2#4]
When I call on GroundedSAM2 with the ontology, it reinstalls SAM2 every time I run the program. Is there some sort of flag I can set that would stop that behavior? It will reinstall SAM2 with the same version even if I have previously run it moments ago.

**Comment by MC-devel-staudt** [comment]
Same issue. I get,

fatal: destination path 'segment-anything-2' already exists and is not an empty directory.

And then it proceeds to reinstall. There should probably be a simple check in GroundedSAM2 from autodistill_grounded_sam_2 that needs to be changed if the library was already built.

**Comment by nihirv** [comment]
in `helpers.py` there's this check which mistakenly always performs a re-installation:
```
    if not os.path.isdir("~/.cache/autodistill/segment_anything_2/segment-anything-2"):
        subprocess.run(
            [
                "git",
                "clone",
                "https://github.com/facebookresearch/segment-anything-2.git",
            ]
        )

        os.chdir("segment-anything-2")

        subprocess.run(["pip", "install", "-e", "."])
```

This is called in `grounded_sam_2.py`.

Simplest workaround is to copy the grounded_sam_2 file to your local workspace and also create your own load_SAM() function which removes the check (or has the correct check)

This also has the benefit of you being able to control which SAM model to use!

**Comment by octavflorescu** [comment]
```    if not os.path.exists(os.path.expanduser("~/.cache/autodistill/segment_anything_2/segment-anything-2")):```

**Comment by walternat1ve** [comment]
yeah, this is really in bad shape. already the installation was not straight forward as expected. had to install many dependencies manually.

**Comment by pwernette** [comment]
Has there been any update on this issue? I recently installed autodistill and have been trying to use Grounded SAM 2 but _still_ appear to be having this issue. Whatever the supposed fix was does not appear to have taken. Using WSL2 on Windows 11 with GPU configured.
![Screenshot 2024-11-19 094113](https://github.com/user-attachments/assets/5b99efdb-f011-4c86-92fe-320b100dc5d0)
